src/Microsoft.PowerShell.Commands.Management/commands/management/Computer.cs

```diff
@@ -2019,35 +2019,25 @@ internal static void WriteNonTerminatingError(int errorcode, PSCmdlet cmdlet, st
         /// <returns></returns>
         internal static bool IsComputerNameValid(string computerName)
         {
-            bool allDigits = true;
+            bool hasNonDigit = false;
 
             if (computerName.Length >= 64)
                 return false;
 
             foreach (char t in computerName)
             {
-                if (t >= 'A' && t <= 'Z' ||
-                    t >= 'a' && t <= 'z')
+                if (t is '-' || char.IsAsciiLetter(t))
```

Comment:
```suggestion
                if (char.IsAsciiLetter(t) || t == '-')
```

Reply:
The computer name usually consists of letters and numbers. The original code clearly demonstrates this by putting the dash check in the last place.